packages/qwik/src/core/shared/qrl/qrl-class.ts

```diff
@@ -113,6 +113,19 @@ export const createQRL = <TYPE>(
       context.$event$ ||= this as Event;
       try {
         return invoke.call(this, context, symbolRef as any, ...(args as any));
+      } catch (e) {
+        if (isDev) {
+          if (isServer) {
+            if (
+              e instanceof ReferenceError &&
+              ['window', 'document'].some((v) => e.message.includes(v))
+            ) {
+              e.message =
+                'It seems like you forgot to add "if (isBrowser) {...}" here: ' + e.message;
+              throw e;
+            }
+          }
+        }
```

Comment:
after here you have to still throw. Best is to move line 125 here so all if cases always throw